tests/unit_tests/test_array_labels.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+import pytest
+
+from tests.utils import load_process_code, run_process_with_additional_js_code
+
+
+@pytest.fixture
+def array_labels_process_code():
+    return load_process_code("array_labels")
+
+
+@pytest.mark.parametrize(
+    "example_input,expected_output",
+    [
+        ({"data": [1, 0, 3, 2], "labels": []}, []),
+        ({"data": [1, 0, 3, 2], "labels": [1, 2, 3, 4]}, [1, 2, 3, 4]),
+        ({"data": [1, 0, 3, 2], "labels": ["1", "2", "3", "4"]}, ["1", "2", "3", "4"]),
+    ],
+)
+def test_array_labels(array_labels_process_code, example_input, expected_output):
+    additional_js_code_to_run = (
+        f"const d = {json.dumps(example_input['data'])};"
+        + f"d.labels = {json.dumps(example_input['labels']) if 'labels' in example_input else 'undefined'};"
+    )
+    process_arguments = f"{{...{json.dumps(example_input)}, 'data': d}}"
+    output = run_process_with_additional_js_code(
+        array_labels_process_code,
+        "array_labels",
+        process_arguments,
+        additional_js_code_to_run,
+    )
+    output = json.loads(output)
+    assert output == expected_output
+
+
+@pytest.mark.parametrize(
+    "example_input,raises_exception,error_message",
+    [
+        ({"data": [1, 0, 3, 2], "labels": [1, 2, 3, 4]}, False, None),
+        ({"data_fake": [1, 0, 3, 2]}, True, "Mandatory argument `data` is not defined."),
+        ({"data": "[1,0,3,2]"}, True, "Argument `data` is not an array."),
+        ({"data": [1, 0, 3, 2]}, True, "Argument `data` is not a labeled array."),
+        ({"data": [1, 0, 3, 2], "labels": "[1,2,3,4]"}, True, "Labels in argument `data` is not an array."),
+    ],
+)
+def test_array_labels_exceptions(array_labels_process_code, example_input, raises_exception, error_message):
+    additional_js_code_to_run = (
+        f"const d = {json.dumps(example_input['data'] if 'data' in example_input else 'undefined')};"
+        + f"d.labels = {json.dumps(example_input['labels']) if 'labels' in example_input else 'undefined'};"
+    )
+    process_arguments = f"{{...{json.dumps(example_input)}, 'data': d}}"
+    if raises_exception:
+        with pytest.raises(Exception) as exc:
+            run_process_with_additional_js_code(
```

Comment:
not sure why new function was added instead of using run_process( array_labels_process_code+ additional_js_code_to_run,...)?

Reply:
At first this function `run_process_with_additional_js_code` was a bit different, more complex (and wrong) and it then got simplified which results in almost the same function as the original `run_process`. the main difference is that the `process_arguments` in `run_process_with_additional_js_code` is a string which contains a JS object and it can not be passed through in any other way than this, meanwhile the `example_input` in `run_process` is a python dictionary (object). So I see only 2 solutions to this problem:
- leave things as they are (not ideal)
- go through all the unit tests and replace so that `json.dumps(example_input)` (which afaik is a string) gets passed through to the `run_process` function so the `run_process_with_additional_js_code` is not needed anymore

Reply:
Perhaps the third option could be that run_process` would accept both, dict and string and than check `example_input` type and convert it if needed.

Reply:
Fixed